include/dr/mhp/algorithms/for_each.hpp

```diff
@@ -53,4 +53,12 @@ void for_each(DI first, DI last, auto op) {
   mhp::for_each(rng::subrange(first, last), op);
 }
 
+/// Collective for_each on iterator/sentinel for a distributed range
+template <dr::distributed_iterator DI, std::integral I>
+void for_each_n(DI first, I n, auto op) {
+  auto last = first;
+  rng::advance(last, n);
```

Comment:
What happens if `n` is greater than range size? Please add a test for this case.

Reply:
> What happens if `n` is greater than range size? Please add a test for this case.

agreed it is ub